xdsl/dialects/riscv.py

```diff
@@ -48,19 +49,66 @@
 from xdsl.utils.hints import isa
 
 
-@dataclass(frozen=True)
-class Register:
+class RISCVRegisterAttr(Data[str], TypeAttribute, ABC):
```

Comment:
I don't really see the point of removing the `Type` suffix, as it is still a type (it inherits from `TypeAttribute`). The argument that it is used as an attribute seems a bit moot to me, as types can be used as attributes anyway...

Reply:
It's a bit arbitrary but somehow feels right to me... I'll think about this some more to try and formulate a rationale.

Reply:
Thought about this some more and it's not a good enough reason for a rename, I changed it back.